### PR TITLE
Fill in FeatureCheckDataflow.cs assertions with issue links

### PR DESCRIPTION
**Pull request by jtschuster**
Adds issues links / explanations to FeatureCheckDataflow, and a couple other places.

I couldn't find the issue for the linker and NativeAOT not being able to handle more complex boolean operations with feature switches. Please let me know if you know if the issue does / doesn't exist and I'll link it / make one.

### CODE REVIEW

File: src/tools/illink/test/Mono.Linker.Tests.Cases/DataFlow/FeatureCheckDataFlow.cs [hunk 1]
```diff
@@ -337,7 +337,7 @@ static void And ()
 			}
```
Comment (jtschuster) [inline src/tools/illink/test/Mono.Linker.Tests.Cases/DataFlow/FeatureCheckDataFlow.cs:337]: https://github.com/dotnet/runtime/pull/102679/files#diff-6283f7ce8ea2a4569a8265503cdd55671a790101ca3c0fd4abd6110c605e30eeL326-L337

This is like the ones I couldn't find an issue for.

Comment (jtschuster) [reply src/tools/illink/test/Mono.Linker.Tests.Cases/DataFlow/FeatureCheckDataFlow.cs:337]: Made https://github.com/dotnet/runtime/issues/102830 to track unifying behavior on this.

**Review by sbomer — APPROVED** [review]
Thank you!

**Comment by jtschuster** [comment]
/ba-g This only changes trimmer tests. Trimmer tests are passing and failures are unrelated.

**Pull request merged**